src/sentry/static/sentry/app/components/buttons/button.jsx

```diff
@@ -89,12 +102,15 @@ class Button extends React.Component {
       'button-disabled': disabled,
     });
 
-    // This container is useless now, but leaves room for when we need to add
-    // components (i.e. icons, busy indicator, etc)
     let childContainer = (
-      <FlowLayout truncate={false}>
-        <span className="button-label">{children}</span>
-      </FlowLayout>
+      <Flex align="center" className="button-label">
+        {icon && (
+          <Icon size={size}>
+            <InlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
```

Comment:
future: I think eventually we'll want icons to be a collection of components and to have them all in storybook would be awesome.

Reply:
Agreeeed. Maybe this is something @Chrissy might be interesting in taking on?